### ISSUE TITLE
Webdav Problem

### ISSUE THREAD

**Issue by morty80** [zhanghai/MaterialFiles#1331]
Hello,

When i add my Wevdav Account to Material Files i can upload Files without any Problems.
But when i copy same Files back to Internal Storage the Files are corrupt and only have a Filesize of 671kB.
Can you please check and fix this Bug?

Best Regards

**Comment by ddubnyk** [comment]
Yeah, same problem here. Unfortunately it's a deal breaker for me as well as I rely heavily on WebDAV for a lot of my setup.

**Comment by Ke1i** [comment]
I successfully managed to add my ownCloud (OCIS version) account to Material files and could list the files and folders without any problem.
Unfortunately I can't create or copy files using Material files - HTTP 400 Bad Request error.
Folder creation works though.
I tried the same using Cx Explorer and it works just fine so Material files is definitely the culprit.

**Comment by morty303** [comment]
> I successfully managed to add my ownCloud (OCIS version) account to Material files and could list the files and folders without any problem. Unfortunately I can't create or copy files using Material files - HTTP 400 Bad Request error. Folder creation works though. I tried the same using Cx Explorer and it works just fine so Material files is definitely the culprit.
>

I also successfully can connect via webdab to my cloud.
Downloads to  Phone works.
Upload from Material Files to Cloud ends with 0KB. :(

**Comment by ddubnyk** [comment]
I've found a workaround in the meantime. If you use DAVx5, you can configure the WebDAV folder through that. Then in material files add an external mount, and choose your DAVx5 Mount. Haven't had any file corruption issues this way.

**Comment by ddubnyk** [comment]
> But @ddubnyk do you get slow transfer speeds with that method? I find it unusable and it's been reported on #711.
> My problem connecting to WebDAV through Material Files' implementation is an error 401 (unauthorized) out of nowhere.

My speeds Max out my Wi-Fi connection speed on the LAN ( over 800Mbps) when copying large files. So I'd say not slow at all.

Over WAN, no noticeable difference for me from any other method.

**Comment by pablogila** [comment]
Interesting. Maybe it's related with DAVx5?
I finally somehow managed to connect the WebDAV server, but only by using my app token as a password. Also, it's giving me an Error every time I try to upload something. Download works properly. For reference, both ways work from my Linux PC.

**Comment by ddubnyk** [comment]
What's your WebDAV server software? Maybe the issue lies there? I simply use Caddy as my WebDAV server.

My current setup is WebDAV Mount in DAVx5. Then add that External Mount to Material files. All has been working very well with this.

If I setup WebDAV directly in Material Files I had corruption issues.

**Comment by ddubnyk** [comment]
Yeah personally I've never had issues with other devices. Other apps on mobile I've experienced issues. 

The only app on mobile I never had a WebDAV issue with while setting it up natively in the app was CX Explorer. 

It seems Android has different ways of handling WebDAV among apps which causes odd issues.
